lighthouse-core/lib/page-functions.js

```diff
@@ -155,15 +155,33 @@ function getOuterHTMLSnippet(element, ignoreAttrs = [], snippetCharacterLimit =
     for (const attributeName of clone.getAttributeNames()) {
       if (charCount > snippetCharacterLimit) {
         clone.removeAttribute(attributeName);
-      } else {
-        let attributeValue = clone.getAttribute(attributeName);
-        if (attributeValue === null) continue;
-        if (attributeValue.length > ATTRIBUTE_CHAR_LIMIT) {
-          attributeValue = attributeValue.slice(0, ATTRIBUTE_CHAR_LIMIT - 1) + '…';
-          clone.setAttribute(attributeName, attributeValue);
+        continue;
+      }
+
+      let attributeValue = clone.getAttribute(attributeName);
+      if (attributeValue === null) continue; // Can't happen.
+
+      let dirty = false;
+
+      // Replace img.src with img.currentSrc.
+      if (attributeName === 'src' && element.tagName === 'IMG') {
+        const currentSrc = /** @type {HTMLImageElement} */ (element).currentSrc;
+        // Only replace if the two URLs do not resolve to the same location.
+        const documentHref = element.ownerDocument.location.href;
+        if (new URL(attributeValue, documentHref).toString() !== currentSrc) {
+          attributeValue = currentSrc;
+          dirty = true;
         }
-        charCount += attributeName.length + attributeValue.length;
       }
+
+      // Elide attribute value if too long.
+      if (attributeValue.length > ATTRIBUTE_CHAR_LIMIT) {
+        attributeValue = attributeValue.slice(0, ATTRIBUTE_CHAR_LIMIT - 1) + '…';
```

Comment:
possible future change: I'm thinking eliding is more useful if done m[i](https://www.youtube.com/watch?v=0rhdOt9bOHE)ddle-out. @Beytoven wdyt?

Reply:
By middle-out do you mean having the '...' in the middle with `ATTRIBUTE_CHAR_LIMIT/2` characters on each side? If so, that could be more useful although, just like the current implementation, the usefulness will be dependent on the element and the way it's attributes are set up.

Reply:
the argument is straightforward for url values (the last bit has the filename, so that's pretty important). but pretty much agree it's 50/50 on which is better for arbitrary values.

there's also the super complex `Util.getURLDisplayName` that does a url-specific eliding

Reply:
additional eliding thought, always elide the origin if it matches the ownerDocument's